iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp

```diff
@@ -71,22 +87,23 @@ std::shared_ptr<CachedQueryAdaptor> QueryAdaptorCache::TryGet(Utf8CP ecsql, bool
         }
         entry->GetStatement().Reset();
         entry->GetStatement().ClearBindings();
+
+        if (ignoreCache)
+            {
+            // Experimental Features are disabled, so finalize and re-prepare the statement to handle error

```

Comment:
This comment is more misleading than useful. IgnoreCache is in the current code only true, if experimental features are disabled. But that doesn't mean we should conclude experimental features are disabled because the flag is true.
Can you explain here, why we need to reprepare the statement if experimental features are not enabled?

Reply:
The need for this re-prepare arises from the following example scenario:
1. An instance query is executed with experimental features enabled and it executes successfully.
2. The same query is called again but this time with experimental features disabled.
In  this scenario, the cached adaptor will be picked up and the already prepared statement will be executed successfully.
We expect a failure in this scenario. Hence the re-prepare, since that's where the experimental features are checked.

Fair point though, the comment and arg name is a bit misleading.

Reply:
Also, the function TryGet() is only called form one place. So the ignoreCache arg being passed is based on the experimental features status. I can change the arg name to be a bit more specific about experimental features and not so generic as ignoreCache.

Reply:
Ah now I get it. And the opposite case does not need handling because if the request failed to prepare, it would not have been added to the cache, so we don't have to bother if they were turned on, only if they were turned off.
Please document this in the code, it's not obvious :-)